Sources_v3/Query/ChannelQuery.swift

```diff
@@ -25,6 +27,18 @@ public struct ChannelQuery<ExtraData: ExtraDataTypes>: Encodable {
     public let options: QueryOptions
     /// ChannelCreatePayload that is needed only when creating channel
     let channelPayload: ChannelEditDetailPayload<ExtraData>?
+    
+    /// ChannelId this query handles.
+    var cid: ChannelId {
+        guard let id = id else { return .init(type: type, id: "") }
```

Comment:
Doesn't this produce an invalid cid? Isn't that exactly what we wanted to avoid having? IMO this should be public and optional - or maybe I just don't understand what you meant by this 🤷‍♂️

Reply:
From my perspective idea was to remove "invalid" cid from the `ChannelId` implementation. 
We can't control if id is valid from the point of` id` string inside `ChannelId`.
Here we have 2 options, right: optional or some dummy id. Making it optional will bring another complexity to already complicated `ChannelController` implementation, ton of new `ifs/guards` so I decided to go with dummy id.
We can make it optional but I'm not sure it's critical there. 
WDYT?

Reply:
I think it is critical tbh. Imagine this:
```
let controller1 = client.channelController(createDirectMessageChannelWith: members)
let controller2 = client.channelController(for: controller1.cid)
```
It's a completely valid thing to do and the consumer of the api has no ida, the cid is not a valid cid.

Reply:
I think we can get rid of `isChannelAlreadyCreated` internally and replace it with `guard let cid`?

Reply:
I see, right.
So I made it optional on `ChannelController` and now we have points to make `cid` public.

Reply:
@mantergo But `ChannelQuery.cid` should be optional, too.